File: 2598-ShortestDistanceToTargetStringInACircularArray/2598-ShortestDistanceToTargetStringInACircularArray.js
// Last updated: 4/29/2026, 10:20:05 AM
/**
 * @param {string[]} words
 * @param {string} target
 * @param {number} startIndex
 * @return {number}
 */
var closestTarget = function(words, target, startIndex) {
    let currIndex = startIndex;
    const n = words.length;
    let steps = 0;
    while(steps < n) {
        const index1 = (startIndex - steps + n) % n;
        const index2 = (startIndex + steps) % n;
        console.log(`steps: ${steps}`)
        console.log(`index1: ${index1}`)
        console.log(`index2: ${index2}`)
        const word1 = words[index1];
        const word2 = words[index2];
        if(word1 === target || word2 === target) {
            return steps;
        }
        steps++;
    }
    return -1;
};

File: 2598-ShortestDistanceToTargetStringInACircularArray/2598-ShortestDistanceToTargetStringInACircularArray.py
# Last updated: 4/29/2026, 10:20:10 AM
class Solution:
    def closetTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        i1 = startIndex
        i2 = startIndex
        count = 0
        maxCount = n // 2 + 1
        while count < maxCount:
            if words[i1] == target or words[i2] == target:
                return count
            i1 = (i1+1) % n
            i2 = (i2-1+n) % n
            count += 1
        return -1
            